pkg/controller/infrastructure/infraflow/shared/whiteboard.go

```diff
@@ -258,10 +274,25 @@ func IsValidValue(value string) bool {
 	return value != "" && value != deleted
 }
 
-// ValidValue returns the value if the exported value is valid, otherwise an empty string.
-func ValidValue(value string) string {
-	if !IsValidValue(value) {
-		return ""
-	}
-	return value
+// HasObject returns true if the object with the given key exists.
+func (w *whiteboard) HasObject(key string) bool {
+	w.Lock()
+	defer w.Unlock()
+	_, ok := w.objects[key]
+	return ok
+}
```

Comment:
I don't object to doing it this way, but checking for the existence of a key should be possible without locking the datastructure. We don't actually gain anything by locking here.

Reply:
This is copied over from the other shared packages. Also not used in this provider at all. But regarding the access, I think that if you use mutexes then you should lock everything, or you can implement it with RWLocks. We either way only lock access to the map and not the objects.

Reply:
> This is copied over from the other shared packages.
Yeah, I figured. Maybe we can find the time to go over the whiteboard for the second round of refactoring, but it's not actually causing errors so it's fine either way.
>  But regarding the access, I think that if you use mutexes then you should lock everything […]
Read-only access to a shared datastructures in general should not require locks. Is there a case where _not_ locking could  cause in a race condition? I can't think of one (that is not also possible with locking, that is). I'll have to read up on go-maps at some point.